cryptographic_estimators/helper.py

```diff
@@ -121,57 +121,80 @@ def is_prime_power(n):
         sage: is_prime_power(11)
         True
 
+        sage: is_prime_power(7^3, return_pair = True)
+        (True, (7, 3))
+
     TESTS::
 
-        sage: is_prime_power(557)
+        sage: is_prime_power(101^2)
         True
 
         sage: is_prime_power(7^3+1)
         False
 
-        sage: is_prime_power(11^4)
-        True
-
         sage: is_prime_power(1121)
         False
 
+        sage: is_prime_power(1087*1091)
+        False
+
     """
     global PRIMES
 
+    if n <= 1:
+        return (False, (None, None)) if return_pair else False
+
     def is_power(n, p):
+        m = 0
         while n % p == 0:
             n //= p
-        return n == 1
-
-    if n <= 1:
-        return False
+            m += 1
+        return (n == 1, m)
+
+    def check_small_primes(n, max_prime):
+        if n < max_prime:
+            index = bisect_left(PRIMES, n)
+            if index < len(PRIMES) and PRIMES[index] == n:
+                return (True, (n, 1)) if return_pair else True
+        for prime in PRIMES:
+            if prime * prime > n:
```

Comment:
Do we really need this condition?

Reply:
Sorry for the delay, I think I already answer this.

Yes, after a carefully review with professor Daniel, everything looks fine. This will only work as an early return, and don't introduce any failure in the results.